### PR TITLE
Better docker support

### PR DESCRIPTION
**Pull request by Saturn745**
This PR simplifies the docker image by having it build the binary and run the built binary at runtime instead of building it at runtime like it was previously. This also includes a GitHub action to automatically build the image and publish it to Github container repo (Can be changed to Dockerhub) on each push (Can change to when a release is made if wanted).

### CODE REVIEW

**Comment by tobychui** [comment]
Hi @Saturn745 , thanks for the PR!
@PassiveLemon , you mind take a look at this if you got some free time this weekend?

**Comment by PassiveLemon** [comment]
I've been thinking about an approach like this but I've been busy with other things. This does mostly envision how I would want it designed but the way it's currently implemented is missing some stuff, most notably being the argument pass-through to the binary when being executed. This limits potential runtime capabilities like changing the management port.

I do intend on implementing this myself though so I won't currently merge. I'll leave this on the table though, in case I want to come back to it.

**Comment by Saturn745** [comment]
> I've been thinking about an approach like this but I've been busy with other things. This does mostly envision how I would want it designed but the way it's currently implemented is missing some stuff, most notably being the argument pass-through to the binary when being executed. This limits potential runtime capabilities like changing the management port.
> 
> I do intend on implementing this myself though so I won't currently merge. I'll leave this on the table though, in case I want to come back to it.

You should be able to pass through the arguments similar to how it was before by using -c ARGS if your using the docker cli or the command key in docker compose

**Comment by PassiveLemon** [comment]
I was not aware that the --entrypoint flag was a thing! In that case, this does look okay. I would have asked to add the notifier back in but I don't think that's really necessary tbh. Just make sure to update the Docker README documentation for that --entrypoint flag.

@tobychui Could you make a Docker account/repo for this repository and create the secret? Trying to manage the keys for mine while keeping them private to me would be a hassle with the workflow from this PR.

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,22 +1,17 @@
...
 
-RUN apk add --no-cache bash curl jq git sudo
+WORKDIR /etc/zoraxy/
 
-RUN mkdir -p /zoraxy/source/ &&\
-    mkdir -p /zoraxy/config/
+VOLUME [ "/etc/zoraxy/config/" ]
 
-VOLUME [ "/zoraxy/config/" ]
+COPY ./src ./
```
Comment (PassiveLemon) [inline docker/Dockerfile:7]: Does this properly get the src directory from the repositories root? I don't have time to test it right now but this seems off to me

**Comment by Saturn745** [comment]
> I was not aware that the --entrypoint flag was a thing! In that case, this does look okay. I would have asked to add the notifier back in but I don't think that's really necessary tbh. Just make sure to update the Docker README documentation for that --entrypoint flag.
> 
> @tobychui Could you make a Docker account/repo for this repository and create the secret? Trying to manage the keys for mine while keeping them private to me would be a hassle with the workflow from this PR.

Great!

File: docker/Dockerfile (hunk 1 above)
Comment (Saturn745) [reply docker/Dockerfile:7]: Yes it does as long as you set the context to the root directory when you go to build the image

**Comment by Saturn745** [comment]
I have updated the Docker README in [`5b44c39`](https://github.com/Saturn745/zoraxy/commit/5b44c390813719392d50c5d0bfa84a5218ab28df) and [`2e18146`](https://github.com/Saturn745/zoraxy/commit/2e18146ebaa16a92836dd8c6d6453191683d4d92)

**Comment by PassiveLemon** [comment]
Looks good to me. I'll get this merged once we get the Docker repo situated

**Comment by tobychui** [comment]
> I was not aware that the --entrypoint flag was a thing! In that case, this does look okay. I would have asked to add the notifier back in but I don't think that's really necessary tbh. Just make sure to update the Docker README documentation for that --entrypoint flag. I will also change the workflow to one of my own to fulfill some specific needs so you could remove that.
> 
> @tobychui Could you make a Docker account/repo for this repository and create the secret for the access key? Or I can do it and just send you the key privately. Trying to manage the keys for mine while keeping them private to me would be a hassle with the workflow from this PR.

Sorry for the late reply. I am currently in travel and I will PM you for the access key after I got back home.

**Comment by PassiveLemon** [comment]
I would have liked to just merge your PR, however I wanted to keep some of my practices in the Dockerfile and trying to communicate what I would like you to add/change would be a hassle so I just redid the whole thing while using your PR as a general guide

**Comment by tobychui** [comment]
> I would have liked to just merge your PR, however I wanted to keep some of my practices in the Dockerfile and trying to communicate what I would like you to add/change would be a hassle so I just redid the whole thing while using your PR as a general guide

@PassiveLemon  I think in the case, we should also let @Saturn745 be inside our list of contributors as you have referenced his PR and update your own. Maybe you could find a way to somehow merge something from his repo / ask him to add something like a comment so he can also be proud of his contribution to Zoraxy?  

Thanks for both of your contributions to make this project better!!!

**Comment by PassiveLemon** [comment]
Yeah Saturn if you want to make a PR and put a comment somewhere, feel free! We would love to add you to the contributors list

**Comment by Saturn745** [comment]
> Yeah Saturn if you want to make a PR and put a comment somewhere, feel free! We would love to add you to the contributors list

Appreciate that but I think I'll find something else to try and contribute in this project.

**Pull request closed without merging**